File: problem152.c
#include <stdio.h>



int number_of_ways(int n, int max, float target, float sum) {
	if(sum == target) {
		printf("Found SUM!\n");
		return 1;
	}

	if(sum > target)
		return 0;

	printf("%f\n", sum);
	
	int c = 0;
	int i;
	for (i = n+1; i <= max; i++) {
		c += number_of_ways(i, max, target, sum + 1.0f / (i*i));
		printf("%i\n", i);
		c++;
		if(c == 2)
			return 0;
	}

	return c;
}

main() {
	int val = number_of_ways(2, 35, 0.5, 0);
	printf("%i\n", val);
}

File: problem152.rb

def number_of_ways(n, remainder, inverses)
	#puts remainder
	if remainder == 0
		puts "Found SUM!"
		return 1
	end

	if remainder < 0
		#puts "too small!"
		return 0
	end

	c = 0
	for i in n...inverses.length
		new_remainder = remainder - inverses[i]
		#next if new_remainder < 0
		c += number_of_ways(i+1, new_remainder, inverses);
	end
	c
end

half_lcm = lcm / 2

inverses = Array.new
for i in 2..50
	next if i == 13
	next if i == 11
	next if i == 17
	next if i == 19
	next if i == 21
	next if i == 22
	next if i == 23
	next if i == 26
	next if i == 29
	next if i == 31
	next if i == 33
	next if i == 34
	puts i
	inverses << (lcm / (i*i))
end
puts inverses
puts number_of_ways(0, half_lcm, inverses)
